### PR TITLE
Feature/37595 add two new anomaly report detections

### PR DESCRIPTION
**Pull request by jon-shipley**
Add two new anomaly report detections:

* Pupil prefs out of sequence
* Duplicates

Remove eslint packages from admin dev dependencies - they don't work for monorepos

### CODE REVIEW

File: admin/services/data-access/retry-async.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict'
 
 const logger = require('../log.service').getLogger()
-const pause = (duration) => new Promise(res => setTimeout(res, duration), noReject => undefined)
+const pause = (duration) => new Promise(resolve => setTimeout(resolve, duration))
```
Comment (GuyHarwood) [inline admin/services/data-access/retry-async.js]: this is a change i made in the retry-async PR - do you need to refresh from master?

Comment (GuyHarwood) [reply admin/services/data-access/retry-async.js]: looks like it's a duplicate change, because i see a conflict with master on this file

Comment (jon-shipley) [reply admin/services/data-access/retry-async.js]: Yes, I made the same change as you.  Now resolved. 😁

File: admin/server.js [hunk 2]
```diff
@@ -67,11 +67,9 @@ function onError (error) {
     case 'EACCES':
       console.error(bind + ' requires elevated privileges')
       process.exit(1)
-      break
     case 'EADDRINUSE':
       console.error(bind + ' is already in use')
       process.exit(1)
-      break
```
Comment (GeorgeChatzigiannis) [inline admin/server.js]: Isn't the linter complaining about this being removed?

Comment (jon-shipley) [reply admin/server.js]: No.  Checked with standard v14.3.1, and v12.0.1

**Pull request merged**